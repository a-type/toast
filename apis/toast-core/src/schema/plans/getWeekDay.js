import { addDays, addWeeks } from 'date-fns';

export default ({ weekIndex, startDay, dayOffset }) => {
  const date = new Date(startDay.valueOf());
  return addDays(addWeeks(date, weekIndex), dayOffset);
};
