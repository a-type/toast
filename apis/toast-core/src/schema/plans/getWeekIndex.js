export default ({ year, month, date, startDay }) => {
  const day = new Date(year, month, date);
  const difference = day - startDay;
  return Math.floor(difference / 604800000);
};
