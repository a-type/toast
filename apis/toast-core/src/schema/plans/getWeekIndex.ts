export default ({
  year,
  month,
  date,
  startDay,
}: {
  year: number;
  month: number;
  date: number;
  startDay: Date;
}) => {
  if (!year || !month || !date || !startDay) {
    throw new Error('All parameters are required');
  }

  const day = new Date(year, month, date);
  const difference = day.valueOf() - startDay.valueOf();
  return Math.floor(difference / 604800000);
};
