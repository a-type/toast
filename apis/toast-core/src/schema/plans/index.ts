import { gql } from 'apollo-server-express';
import { Context } from 'context';
import { path, mergeDeepRight } from 'ramda';
import { id } from 'tools';

import * as meals from './meals';
import * as schedules from './schedules';

export const typeDefs = () => [
  gql`
    type Plan {
      id: ID!
    }

    extend type Query {
      plan: Plan
    }
  `,
  meals.typeDefs,
  schedules.typeDefs,
];

export const resolvers = [
  {
    Query: {
      plan: async (_parent, args, ctx: Context) => {
        const group = await ctx.graph.groups.getMine();
        let planId = path<string>(['planId'], group);

        if (!planId) {
          planId = id('plan');
          await ctx.graph.groups.mergeMine({ planId });
        }

        const plan = await ctx.firestore.plans.get(planId);

        return plan;
      },
    },
  },
  meals.resolvers,
  schedules.resolvers,
].reduce(mergeDeepRight, {});
