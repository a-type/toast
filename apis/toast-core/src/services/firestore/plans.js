const COLLECTION = 'plans';
const START_WEEK_DAY = new Date(2018, 10, 7);

const defaulted = plan => ({
  servingsPerMeal: 0,
  days: [],
  ...plan,
});

export default class Plans {
  static START_WEEK_DAY = START_WEEK_DAY;
  START_WEEK_DAY = START_WEEK_DAY;

  constructor(firestore) {
    this.firestore = firestore;
  }

  /**
   * For user's default plan, use user id
   */
  get = async planId => {
    const document = this.firestore.doc(`${COLLECTION}/${planId}`);
    const docRef = await document.get();
    if (!docRef.exists) {
      return null;
    }

    return defaulted(docRef.data());
  };

  merge = async (planId, planData) => {
    const document = this.firestore.doc(`${COLLECTION}/${planId}`);
    await document.set(planData, { merge: true });
    const docRef = await document.get();
    return defaulted(docRef.data());
  };

  set = async (planId, plan) => {
    const document = this.firestore.doc(`${COLLECTION}/${planId}`);
    await document.set(plan);
    return defaulted(plan);
  };

  /**
   * Weeks - copies of the base plan, applied per-week
   */
  getWeek = async (planId, week) => {
    const document = this.firestore.doc(`${COLLECTION}/${planId}/week_${week}`);
    const docRef = await document.get();

    if (!docRef.exists) {
      // copy the base plan instead
      return await this.get(planId);
    }

    return defaulted(docRef.data());
  };

  mergeWeek = async (planId, week, planData) => {
    const document = this.firestore.doc(`${COLLECTION}/${planId}/week_${week}`);
    await document.set(planData, { merge: true });
    const docRef = await document.get();
    return defaulted(docRef.data());
  };
}
