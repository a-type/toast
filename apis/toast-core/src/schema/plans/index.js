import { gql } from 'apollo-server-express';
import { id } from 'tools';
import { path, pathOr, assocPath, compose, mergeDeepLeft } from 'ramda';
import { UserInputError } from 'errors';
import getWeekIndex from './getWeekIndex';
import getWeekDay from './getWeekDay';
import Plan from 'services/planner/Plan';

const emptyPlan = {
  days: new Array(7).fill(null).map(() => ({
    meals: new Array(3).fill(null).map(() => ({
      availability: 'SKIP',
      actions: [],
    })),
  })),
};

export const typeDefs = gql`
  enum PrepAvailability {
    SKIP
    EAT_OUT
    NONE
    SHORT
    MEDIUM
    LONG
  }

  enum PlanActionType {
    EAT_OUT
    COOK
    EAT
    READY_MADE
    SKIP
  }

  enum PlanMealType {
    QUICK
    BIG
    FANCY
    NORMAL
  }

  enum PlanStrategy {
    BASIC
    PREP
    BIG_PREP
  }

  interface PlanAction {
    id: ID!
    type: PlanActionType!
    dayIndex: Int
    mealIndex: Int
  }

  type PlanActionEatOut implements PlanAction {
    id: ID!
    type: PlanActionType!
    dayIndex: Int
    mealIndex: Int
    note: String
  }

  type PlanActionCook implements PlanAction {
    id: ID!
    type: PlanActionType!
    dayIndex: Int
    mealIndex: Int
    servings: Int!
    mealType: PlanMealType!
    recipe: Recipe
  }

  type PlanActionEat implements PlanAction {
    id: ID!
    type: PlanActionType!
    dayIndex: Int
    mealIndex: Int
    leftovers: Boolean!
    cookAction: PlanActionCook
  }

  type PlanActionReadyMade implements PlanAction {
    id: ID!
    type: PlanActionType!
    dayIndex: Int
    mealIndex: Int
    note: String
  }

  type PlanActionSkip implements PlanAction {
    id: ID!
    type: PlanActionType!
    dayIndex: Int
    mealIndex: Int
  }

  type PlanMeal {
    id: ID!
    availability: PrepAvailability!
    actions: [PlanAction!]!
  }

  type PlanDay {
    id: ID!
    date: Date
    meals: [PlanMeal!]!
  }

  type Plan {
    id: ID!
    defaultServings: Int!
    days: [PlanDay!]!
    groceryDay: Int!
    warnings: [String!]!

    """
    Access weekly plans using the week index, which is based
    on the fixed plan chronology system. To calculate a week index
    for any particular day, use the root planWeekIndex query
    """
    week(weekIndex: Int!): Plan

    """
    A date representing the first day (Sunday) of this Plan week,
    if (and only if) this Plan is a weekly plan, and not the root "blueprint" plan
    """
    startDate: Date
  }

  input PlanSetDetailsInput {
    defaultServings: Int
    groceryDay: Int
  }

  input PlanSetMealDetailsInput {
    availability: PrepAvailability
  }

  extend type Mutation {
    setPlanDetails(details: PlanSetDetailsInput!): Plan!
      @hasScope(scope: "update:plan")
    setPlanMealDetails(
      dayIndex: Int!
      mealIndex: Int!
      details: PlanSetMealDetailsInput!
    ): Plan! @hasScope(scope: "update:plan")
    setPlanStrategy(strategy: PlanStrategy): Plan!

    setPlanActionRecipe(
      weekIndex: Int!
      dayIndex: Int!
      mealIndex: Int!
      actionId: ID!
      recipeId: ID!
    ): PlanAction!
  }

  extend type Group {
    plan: Plan @authenticated
  }

  extend type Query {
    """
    A shortcut function to determine the week index of any particular day
    within the planning system
    """
    planWeekIndex(year: Int!, month: Int!, date: Int!): Int!
    """
    The start date of the entire plan system. You can reference this
    to determine the chronology of plan weeks in conjunction with
    planWeekIndex and the weekIndex field on Plan itself
    """
    planStartWeekDate: Date!
  }
`;

export const resolvers = {
  Query: {
    planWeekIndex: (_parent, { year, month, date }, ctx) =>
      getWeekIndex({
        year,
        month,
        date,
        startDay: ctx.firestore.plans.START_WEEK_DAY,
      }),

    planStartWeekDate: (_parent, _args, ctx) =>
      ctx.firestore.plans.START_WEEK_DAY,
  },

  Mutation: {
    setPlanDetails: async (_parent, { details }, ctx) => {
      const group = await ctx.graph.groups.getMine();
      let planId = path(['planId'], group);
      // create plan if it doesn't exist
      if (!planId) {
        planId = id('plan');
        await ctx.graph.groups.mergeMine({ planId });
      }

      const defaultPlan = new Plan({ id: planId });
      const plan = (await ctx.firestore.plans.get(planId)) || defaultPlan;
      plan.defaultServings = details.defaultServings;
      plan.groceryDay = details.groceryDay;
      const processed = ctx.planner.run(plan);
      return ctx.firestore.plans.set(planId, processed);
    },

    setPlanMealDetails: async (
      _parent,
      { dayIndex, mealIndex, details },
      ctx,
    ) => {
      const group = await ctx.graph.groups.getMine();

      if (!group || !group.planId) {
        throw new UserInputError("You haven't created a plan yet");
      }

      const plan = await ctx.firestore.plans.get(group.planId);

      plan.setMealDetails(dayIndex, mealIndex, details);
      const processed = ctx.planner.run(plan);

      await ctx.firestore.plans.set(group.planId, plan);
      return plan;
    },

    setPlanStrategy: async (_parent, { strategy }, ctx) => {
      const group = await ctx.graph.groups.getMine();

      if (!group || !group.planId) {
        throw new UserInputError("You haven't created a plan yet");
      }

      const plan = await ctx.firestore.plans.get(group.planId);
      plan.strategy = strategy;
      const processed = ctx.planner.run(plan);
      await ctx.firestore.plans.set(group.planId, processed);
      return processed;
    },

    setPlanActionRecipe: async (
      _parent,
      { weekIndex, dayIndex, mealIndex, actionId, recipeId },
      ctx,
    ) => {
      const group = await ctx.graph.groups.getMine();

      if (!group || !group.planId) {
        throw new UserInputError("You haven't created a plan yet");
      }

      const plan = await ctx.firestore.plans.getWeek(group.planId, weekIndex);
      const action = plan.setActionRecipe(
        dayIndex,
        mealIndex,
        actionId,
        recipeId,
      );
      await ctx.firestore.plans.mergeWeek(group.planId, weekIndex, plan);
      return action;
    },
  },

  Group: {
    plan: async (parent, args, ctx) => {
      const { planId } = parent;
      const plan = await ctx.firestore.plans.get(planId);
      ctx.plan = plan;
      return plan;
    },
  },

  Plan: {
    week: async (parent, args, ctx) => {
      if (parent.weekIndex !== null) {
        throw new UserInputError(
          "You can't access the week field on a Plan which represents a week already",
        );
      }
      const week = await ctx.firestore.plans.getWeek(parent.id, args.weekIndex);
      ctx.week = week;
      return week;
    },

    startDate: (parent, _args, ctx) => {
      if (!parent.id.includes('week_')) {
        return null;
      }

      return getWeekDay({
        weekIndex: parent.weekIndex,
        startDay: ctx.firestore.plans.START_WEEK_DAY,
      });
    },
  },

  PlanAction: {
    __resolveType: obj => {
      switch (obj.type) {
        case 'COOK':
          return 'PlanActionCook';
        case 'EAT':
          return 'PlanActionEat';
        case 'EAT_OUT':
          return 'PlanActionEatOut';
        case 'READY_MADE':
          return 'PlanActionReadyMade';
        case 'SKIP':
          return 'PlanActionSkip';
      }
    },
  },

  PlanActionEat: {
    cookAction: async (parent, args, ctx) => {
      let plan = ctx.week || ctx.plan;
      if (!plan) {
        const group = await ctx.graph.groups.getMine();

        if (!group || !group.planId) {
          throw new UserInputError("You haven't created a plan yet");
        }

        plan = await ctx.firestore.plans.get(group.planId);
      }

      const { cookActionId } = parent;

      return plan.getAction(cookActionId);
    },
  },

  PlanActionCook: {
    recipe: async (parent, args, ctx) => {
      if (!parent.recipeId) {
        return null;
      }

      return ctx.graph.recipes.get(parent.recipeId);
    },
  },

  PlanDay: {
    date: (parent, args, ctx) => {
      if (ctx.week) {
        const weekIndex = ctx.week.weekIndex;
        const { dayIndex } = parent;

        const date = getWeekDay({
          weekIndex,
          startDay: ctx.firestore.plans.START_WEEK_DAY,
          dayOffset: dayIndex,
        });
        return date;
      }
      return null;
    },
  },
};
