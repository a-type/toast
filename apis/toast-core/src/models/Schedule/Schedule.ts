import { id } from 'tools';
import Meal, { MealData } from 'models/Meal/Meal';
import planner from './planner';

export enum ScheduleStrategy {
  Basic = 'BASIC',
  Prep = 'PREP',
}

export type ScheduleMeal = {
  id: string;
  mealIndex: number;
  dayIndex: number;
  availability: string; // TODO: enum
  servings?: number;
};

type BaseScheduleData = {
  id?: string;
  defaultServings: number;
  strategy?: ScheduleStrategy;
  meals: ScheduleMeal[];
  warnings: string[];
};

export type ScheduleData = BaseScheduleData & {
  templateWeek?: MealData[];
};

type InternalScheduleData = BaseScheduleData & {
  templateWeek?: Meal[];
};

const createEmpty = (scheduleId): ScheduleData => {
  return {
    id: scheduleId,
    defaultServings: 2,
    strategy: ScheduleStrategy.Basic,

    meals: new Array(21).fill(null).map((__, mealIndex) => ({
      id: `${scheduleId}_m_${mealIndex}`,
      mealIndex,
      dayIndex: Math.floor(mealIndex / 3),
      availability: 'SKIP',
      servings: 2,
    })),

    templateWeek: [],

    warnings: [],
  };
};

export default class Schedule {
  data: InternalScheduleData;
  generateId: (name: string) => string;

  constructor(data?: ScheduleData, { generateId = id } = {}) {
    this.generateId = generateId;
    const incomingData = data || createEmpty(generateId('schedule'));
    this.data = {
      ...incomingData,
      templateWeek: incomingData.templateWeek.map<Meal>(meal => new Meal(meal)),
    };
  }

  get id() {
    return this.data.id;
  }

  get strategy() {
    return this.data.strategy;
  }

  get defaultServings() {
    return this.data.defaultServings;
  }

  get meals() {
    return [...this.data.meals];
  }

  get warnings() {
    return [...this.data.warnings];
  }

  get templateWeek() {
    return [...this.data.templateWeek];
  }

  set id(scheduleId) {
    this.data.id = scheduleId || null;
  }

  set defaultServings(defaultServings) {
    this.data.defaultServings = defaultServings || 2;
    this.updateTemplateWeek();
  }

  set strategy(strategy) {
    this.data.strategy = strategy;
    this.updateTemplateWeek();
  }

  getScheduleMeal = (day: number, meal: number): ScheduleMeal => {
    return this.meals[day * 3 + meal];
  };

  setScheduleMealDetails = (dayIndex, mealIndex, details = {}) => {
    const meal = this.data.meals[dayIndex * 3 + mealIndex];
    Object.assign(meal, details);
    this.updateTemplateWeek();
  };

  updateTemplateWeek = () => {
    this.data.templateWeek = planner.run(this);
  };

  getPlanMeal = (dateIndex: number, mealIndex: number) => {
    // copy basic data from template week and update date
    const dayIndex = dateIndex % 7;
    const templateMeal = this.data.templateWeek.find(
      meal => meal.dayIndex === dayIndex && meal.mealIndex === mealIndex,
    );
    return templateMeal.move(Math.floor((dateIndex - dayIndex) / 7));
  };

  addWarning = warning => {
    this.data.warnings.push(warning);
  };

  reset = () => {
    this.data.warnings = [];
  };

  countMealsWithAvailability = availability => {
    const options = [].concat(availability);
    return this.meals.filter(meal => options.includes(meal.availability))
      .length;
  };

  toJSON() {
    return {
      ...this.data,
      templateWeek: this.data.templateWeek.map(meal => meal.toJSON()),
    };
  }

  static fromJSON(data) {
    return new Schedule(data);
  }

  static createEmpty(id: string) {
    return new Schedule({
      id,
      defaultServings: 2,
      strategy: ScheduleStrategy.Basic,

      meals: new Array(21).fill(null).map((__, mealIndex) => ({
        id: `${id}_m_${mealIndex}`,
        mealIndex,
        dayIndex: Math.floor(mealIndex / 3),
        availability: 'SKIP',
        servings: 2,
      })),

      templateWeek: new Array(21)
        .fill(null)
        .map((_, mealIndex) =>
          Meal.createEmpty(Math.floor(mealIndex / 3), mealIndex % 3),
        ),

      warnings: [],
    });
  }
}
