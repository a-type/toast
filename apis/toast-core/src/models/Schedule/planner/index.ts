import basic from './basic';
import prep from './prep';
import { Schedule, Meal } from 'models';
import { ScheduleStrategy } from 'models/Schedule/Schedule';
import { initializeWeek } from './utils';

export default {
  run: (schedule: Schedule): Meal[] => {
    const meals = initializeWeek();

    switch (schedule.strategy) {
      case ScheduleStrategy.Prep:
        return prep(schedule, meals);
      default:
        return basic(schedule, meals);
    }
  },
};
