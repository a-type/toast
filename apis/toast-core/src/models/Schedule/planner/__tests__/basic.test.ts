import plan from '../index';
import { ScheduleStrategy } from 'models/Schedule/Schedule';
import plans from './createPlans';
const { variety } = plans();

describe('basic meal plan', () => {
  test('converts availability to meal types', () => {
    variety.strategy = ScheduleStrategy.Basic;
    expect(plan.run(variety).map(m => m.toJSON())).toMatchSnapshot();
  });
});
