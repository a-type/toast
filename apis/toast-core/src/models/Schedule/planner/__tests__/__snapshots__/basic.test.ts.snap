// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic meal plan converts availability to meal types 1`] = `
Array [
  Object {
    "actions": Array [
      Object {
        "id": "m_0_0_a_0_SKIP",
        "type": "SKIP",
      },
    ],
    "dateIndex": 0,
    "id": "meal_0_0",
    "mealIndex": 0,
  },
  Object {
    "actions": Array [
      Object {
        "id": "m_0_1_a_0_COOK",
        "recipeType": "QUICK",
        "servings": 2,
        "type": "COOK",
      },
      Object {
        "cookActionId": "m_0_1_a_0_COOK",
        "id": "m_0_1_a_1_EAT",
        "leftovers": false,
        "type": "EAT",
      },
    ],
    "dateIndex": 0,
    "id": "meal_0_1",
    "mealIndex": 1,
  },
  Object {
    "actions": Array [
      Object {
        "id": "m_0_2_a_0_COOK",
        "recipeType": "FANCY",
        "servings": 2,
        "type": "COOK",
      },
      Object {
        "cookActionId": "m_0_2_a_0_COOK",
        "id": "m_0_2_a_1_EAT",
        "leftovers": false,
        "type": "EAT",
      },
    ],
    "dateIndex": 0,
    "id": "meal_0_2",
    "mealIndex": 2,
  },
  Object {
    "actions": Array [
      Object {
        "id": "m_1_0_a_0_SKIP",
        "type": "SKIP",
      },
    ],
    "dateIndex": 1,
    "id": "meal_1_0",
    "mealIndex": 0,
  },
  Object {
    "actions": Array [
      Object {
        "id": "m_1_1_a_0_EAT_OUT",
        "type": "EAT_OUT",
      },
    ],
    "dateIndex": 1,
    "id": "meal_1_1",
    "mealIndex": 1,
  },
  Object {
    "actions": Array [
      Object {
        "id": "m_1_2_a_0_COOK",
        "recipeType": "NORMAL",
        "servings": 2,
        "type": "COOK",
      },
      Object {
        "cookActionId": "m_1_2_a_0_COOK",
        "id": "m_1_2_a_1_EAT",
        "leftovers": false,
        "type": "EAT",
      },
    ],
    "dateIndex": 1,
    "id": "meal_1_2",
    "mealIndex": 2,
  },
  Object {
    "actions": Array [
      Object {
        "id": "m_2_0_a_0_COOK",
        "recipeType": "QUICK",
        "servings": 2,
        "type": "COOK",
      },
      Object {
        "cookActionId": "m_2_0_a_0_COOK",
        "id": "m_2_0_a_1_EAT",
        "leftovers": false,
        "type": "EAT",
      },
    ],
    "dateIndex": 2,
    "id": "meal_2_0",
    "mealIndex": 0,
  },
  Object {
    "actions": Array [
      Object {
        "id": "m_2_1_a_0_COOK",
        "recipeType": "QUICK",
        "servings": 2,
        "type": "COOK",
      },
      Object {
        "cookActionId": "m_2_1_a_0_COOK",
        "id": "m_2_1_a_1_EAT",
        "leftovers": false,
        "type": "EAT",
      },
    ],
    "dateIndex": 2,
    "id": "meal_2_1",
    "mealIndex": 1,
  },
  Object {
    "actions": Array [
      Object {
        "id": "m_2_2_a_0_COOK",
        "recipeType": "QUICK",
        "servings": 2,
        "type": "COOK",
      },
      Object {
        "cookActionId": "m_2_2_a_0_COOK",
        "id": "m_2_2_a_1_EAT",
        "leftovers": false,
        "type": "EAT",
      },
    ],
    "dateIndex": 2,
    "id": "meal_2_2",
    "mealIndex": 2,
  },
  Object {
    "actions": Array [
      Object {
        "id": "m_3_0_a_0_SKIP",
        "type": "SKIP",
      },
    ],
    "dateIndex": 3,
    "id": "meal_3_0",
    "mealIndex": 0,
  },
  Object {
    "actions": Array [
      Object {
        "id": "m_3_1_a_0_SKIP",
        "type": "SKIP",
      },
    ],
    "dateIndex": 3,
    "id": "meal_3_1",
    "mealIndex": 1,
  },
  Object {
    "actions": Array [
      Object {
        "id": "m_3_2_a_0_EAT_OUT",
        "type": "EAT_OUT",
      },
    ],
    "dateIndex": 3,
    "id": "meal_3_2",
    "mealIndex": 2,
  },
  Object {
    "actions": Array [
      Object {
        "id": "m_4_0_a_0_SKIP",
        "type": "SKIP",
      },
    ],
    "dateIndex": 4,
    "id": "meal_4_0",
    "mealIndex": 0,
  },
  Object {
    "actions": Array [
      Object {
        "id": "m_4_1_a_0_SKIP",
        "type": "SKIP",
      },
    ],
    "dateIndex": 4,
    "id": "meal_4_1",
    "mealIndex": 1,
  },
  Object {
    "actions": Array [
      Object {
        "id": "m_4_2_a_0_COOK",
        "recipeType": "NORMAL",
        "servings": 2,
        "type": "COOK",
      },
      Object {
        "cookActionId": "m_4_2_a_0_COOK",
        "id": "m_4_2_a_1_EAT",
        "leftovers": false,
        "type": "EAT",
      },
    ],
    "dateIndex": 4,
    "id": "meal_4_2",
    "mealIndex": 2,
  },
  Object {
    "actions": Array [
      Object {
        "id": "m_5_0_a_0_COOK",
        "recipeType": "FANCY",
        "servings": 2,
        "type": "COOK",
      },
      Object {
        "cookActionId": "m_5_0_a_0_COOK",
        "id": "m_5_0_a_1_EAT",
        "leftovers": false,
        "type": "EAT",
      },
    ],
    "dateIndex": 5,
    "id": "meal_5_0",
    "mealIndex": 0,
  },
  Object {
    "actions": Array [
      Object {
        "id": "m_5_1_a_0_COOK",
        "recipeType": "QUICK",
        "servings": 2,
        "type": "COOK",
      },
      Object {
        "cookActionId": "m_5_1_a_0_COOK",
        "id": "m_5_1_a_1_EAT",
        "leftovers": false,
        "type": "EAT",
      },
    ],
    "dateIndex": 5,
    "id": "meal_5_1",
    "mealIndex": 1,
  },
  Object {
    "actions": Array [
      Object {
        "id": "m_5_2_a_0_COOK",
        "recipeType": "FANCY",
        "servings": 2,
        "type": "COOK",
      },
      Object {
        "cookActionId": "m_5_2_a_0_COOK",
        "id": "m_5_2_a_1_EAT",
        "leftovers": false,
        "type": "EAT",
      },
    ],
    "dateIndex": 5,
    "id": "meal_5_2",
    "mealIndex": 2,
  },
  Object {
    "actions": Array [
      Object {
        "id": "m_6_0_a_0_SKIP",
        "type": "SKIP",
      },
    ],
    "dateIndex": 6,
    "id": "meal_6_0",
    "mealIndex": 0,
  },
  Object {
    "actions": Array [
      Object {
        "id": "m_6_1_a_0_SKIP",
        "type": "SKIP",
      },
    ],
    "dateIndex": 6,
    "id": "meal_6_1",
    "mealIndex": 1,
  },
  Object {
    "actions": Array [
      Object {
        "id": "m_6_2_a_0_COOK",
        "recipeType": "FANCY",
        "servings": 2,
        "type": "COOK",
      },
      Object {
        "cookActionId": "m_6_2_a_0_COOK",
        "id": "m_6_2_a_1_EAT",
        "leftovers": false,
        "type": "EAT",
      },
    ],
    "dateIndex": 6,
    "id": "meal_6_2",
    "mealIndex": 2,
  },
]
`;
